source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -248,15 +248,17 @@ Network::IoResult SslSocket::doWrite(Buffer::Instance& write_buffer, bool end_st
   while (bytes_to_write > 0) {
     // TODO(mattklein123): As it relates to our fairness efforts, we might want to limit the number
     // of iterations of this loop, either by pure iterations, bytes written, etc.
+    const auto slice = write_buffer.maybeLinearize(16384, 4096);
```

Comment:
I recommend setting the copy threshold to 4000 bytes instead of 4096.  This is related to the buffer default slice size being 4096 - sizeof(OwnedSlice) which is about 4032 bytes.  Setting it to 4096 will result in a lot of spurious copies.  See also https://github.com/envoyproxy/envoy/pull/14054#discussion_r526476434